### PR TITLE
BREAKING(fs): remove `Deno.fsync[Sync]()`

### PR DESCRIPTION
**Pull request by iuioiua**
Towards #22079

### CODE REVIEW

File: cli/tsc/dts/lib.deno.ns.d.ts
Comment (iuioiua) [inline cli/tsc/dts/lib.deno.ns.d.ts:1]: These interfaces never had `@deprecated` add to their JSDocs 😬

**Comment by iuioiua** [comment]
Ah, so `Deno.fsync[Sync]()` and `Deno.fdatasync[Sync]()` were previously deprecated. Then the deprecation was reverted in #22085. I think we should deprecate these symbols in Deno 2 and remove them in Deno 2.1.

**Comment by bartlomieju** [comment]
> Ah, so `Deno.fsync[Sync]()` and `Deno.fdatasync[Sync]()` were previously deprecated. Then the deprecation was reverted in #22085. I think we should deprecate these symbols in Deno 2 and remove them in Deno 2.1.

We should remove them in Deno 2, that's the time to make breaking changes.

**Review by bartlomieju — APPROVED** [review]
LGTM

**Pull request merged**